apps/wallet/src/background/Keyring.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Ed25519Keypair } from '@mysten/sui.js';
+import { BehaviorSubject } from 'rxjs';
+import Browser from 'webextension-polyfill';
+
+import { encrypt, decrypt } from '_shared/cryptography/keystore';
+import { generateMnemonic } from '_shared/utils/bip39';
+
+import type { Keypair } from '@mysten/sui.js';
+
+const STORAGE_KEY = 'vault';
+
+class Keyring {
+    private readonly _locked = new BehaviorSubject<boolean>(true);
+    private _encryptedMnemonic: Promise<string | null>;
+    private _keypair: Keypair | null = null;
```

Comment:
Can we use true JS privates instead of underscore-prefixed properties?

Reply:
My goal here was to just differentiate the names here from the getters. But I can change it!